01-node-tutorial/answers/fs-async.js

```diff
@@ -0,0 +1,39 @@
+const { readFile, writeFile } = require("fs");
+
+const line1 =
+  "Async functions are a new feature in JavaScript that allow you to write asynchronous code that is more readable and maintainable.";
+const line2 =
+  "Callbacks are a way of passing a function as an argument to another function to be executed later.";
+const line3 =
+  "Promises are a more advanced way of handling asynchronous code that can help you avoid callback hell.";
+
+console.log("at start");
+
+writeFile("./temporary/fileB.txt", `${line1} \n`, (err, result) => {
+  console.log("at point 1");
+  if (err) {
+    console.log("This error happened: ", err);
+    return;
+  } else {
```

Comment:
You actually do not need the `else` here. The `return` on the line above takes care of this exiting the function if an error happens, so you can write your next statements as normal instead of using the `else`.